[DOCS] Adds 8.2.0 release notes (#130481)
elastic/kibana: docs/CHANGELOG.asciidoc

// tag::notable-breaking-changes[]

There are no breaking changes in {kib} 8.2.0.

// end::notable-breaking-changes[]

To review the breaking changes in previous versions, refer to the following: 

{kibana-ref-all}/8.1/release-notes-8.1.0.html#breaking-changes-8.1.0[8.1.0] | {kibana-ref-all}/8.0/release-notes-8.0.0.html#breaking-changes-8.0.0[8.0.0] | {kibana-ref-all}/8.0/release-notes-8.0.0-rc2.html#breaking-changes-8.0.0-rc2[8.0.0-rc2] | {kibana-ref-all}/8.0/release-notes-8.0.0-rc1.html#breaking-changes-8.0.0-rc1[8.0.0-rc1] | {kibana-ref-all}/8.0/release-notes-8.0.0-beta1.html#breaking-changes-8.0.0-beta1[8.0.0-beta1] | {kibana-ref-all}/8.0/release-notes-8.0.0-alpha2.html#breaking-changes-8.0.0-alpha2[8.0.0-alpha2] | {kibana-ref-all}/8.0/release-notes-8.0.0-alpha1.html#breaking-changes-8.0.0-alpha1[8.0.0-alpha1]
      
[float]
[[features-8.2.0]]
=== Features
{kib} 8.2.0 adds the following new and notable features.

Alerting::
* Keep the number_of_scheduled_actions in event log {kibana-pull}128438[#128438]
* Remove defaultRuleTaskTimeout and set ruleType specific timeout from kibana.yml {kibana-pull}128294[#128294]

> * Limit the executable actions per Rule execution {kibana-pull}128079[#128079] and {kibana-pull}126902[#126902]

Reviewer: ```suggestion
* Limit the executable actions per rule execution {kibana-pull}128079[#128079] and {kibana-pull}126902[#126902]
```

> * Adds Cases to the Stack Management page {kibana-pull}125224[#125224]

Reviewer 2: Maybe Cases should be on it is own category. @peteharverson @sophiec20 what do you think?

Reviewer 3: Yes, having a separate section cases for Cases makes sense to me - Cases aren't just about alerting, and are visible in various places inside Kibana (Security, Observability and now Stack Management).

Dashboard::
* Adds time slider control {kibana-pull}128305[#128305]
* Adds Control group search settings {kibana-pull}128090[#128090]
* Adds hierarchical chaining setting to Controls {kibana-pull}126649[#126649]
* Adds options list API and validation system {kibana-pull}123889[#123889]

Discover::
Adds `Copy to clipboard` ability for column name of Document Explorer {kibana-pull}123892[#123892]

Elastic Security::
For the Elastic Security 8.2.0 release information, refer to {security-guide}/release-notes.html[_Elastic Security Solution Release Notes_].

Lens & Visualizations::

> * Adds manual annotations in *Lens* {kibana-pull}126456[#126456]

Reviewer 4: Should we call out it's a tech preview feature here?

Author: We call out the technical preview in the What's new and main Lens docs.

* Adds multi-field group by in *TSVB* {kibana-pull}126015[#126015]
* Adds ability to navigate to *Lens* with your current *TSVB* configuration {kibana-pull}114794[#114794]

Machine Learning::

> * Anomaly Detection: Add link to maps in charts section of Anomaly Explorer {kibana-pull}128697[#128697]

Reviewer: ```suggestion
* Add link to maps in charts section of Anomaly Explorer {kibana-pull}128697[#128697]
```

* Adds "View in App URL" {{context.viewInAppUrl}} variable to the rule templating language {kibana-pull}127890[#127890]
* Adds view in app url as an action variable in the alert message for uptime app {kibana-pull}127478[#127478]

For more information about the features introduced in 8.2.0, refer to <<whats-new,What's new in 8.2>>.

[[enhancements-and-bug-fixes-v8.2.0]]
=== Enhancements and bug fixes

For detailed information about the 8.2.0 release, review the enhancements and bug fixes.

[float]
[[enhancement-v8.2.0]]
=== Enhancements
Alerting::
* Adds error logs in rule details page {kibana-pull}128925[#128925]
* Simplify error banner on rules {kibana-pull}128705[#128705]
* Adds Previous Snooze button {kibana-pull}128539[#128539]
* Adds Snooze UI and Unsnooze API {kibana-pull}128214[#128214]
* Adds aggs to know how many rules are snoozed {kibana-pull}128212[#128212]
* Adds a connector for xMatters {kibana-pull}122357[#122357]

Dashboard::
* Adds option to open dashboard drilldowns in new tab or window {kibana-pull}125773[#125773]
* Adds range slider Control {kibana-pull}125584[#125584]

Discover::
* Adds ability to edit histogram as vis {kibana-pull}125705[#125705]

> * Enables document explorer as default {kibana-pull}125485[#125485]

Reviewer 5: I wonder if this should be highlighted in a feature, Document explorer has been around for a while, but enabling it as default document table is a huge change

Elastic Security::
For the Elastic Security 8.2.0 release information, refer to {security-guide}/release-notes.html[_Elastic Security Solution Release Notes_].

Enterprise Search::
For the Elastic Enterprise Search 8.2.0 release information, refer to {enterprise-search-ref}/changelog.html[_Elastic Enterprise Search Documentation Release notes_].

Fleet::

> * Adds link to ILM policy documentation {kibana-pull}128656[#128656]

Reviewer 6: Minor, but I think we should use parallel verbs throughout.

Author: Agreed, but I leave the wording up to the area leads.

> * Adds link to ILM policy documentation {kibana-pull}128656[#128656]

Reviewer 6: I'd remove this. I don't think it's link worthy.

> * Adds link to ILM policy documentation {kibana-pull}128656[#128656]

Reviewer 6: ```suggestion
```

* Merge settings & mappings component template in @package {kibana-pull}128498[#128498]
* Redesign agent flyout {kibana-pull}128381[#128381]

> * Adds a link from agent details page to agent dashboard {kibana-pull}127882[#127882]

Reviewer 6: I'd also remove this because I think users will see the change and don't need to read about it in the relnotes.

* Adds ability to open *Lens* visualizations in *Discover* from dashboards {kibana-pull}127355[#127355]
* Auto-set exists filtering for last value in *Lens* {kibana-pull}127251[#127251]
* Adds ability to include number of values in default terms field label in *lens* {kibana-pull}127222[#127222]
* Adds ability to drop partial buckets option in *Lens* {kibana-pull}127153[#127153]
* Addds ability to allow top metric for last value in *Lens* {kibana-pull}127151[#127151]
* Improves Datatable content height with custom row height in *Lens* {kibana-pull}127134[#127134]
* Adds ability to set legend pixel width in *Lens* {kibana-pull}126018[#126018]
* Adds underlying data editor navigation in *Lens* {kibana-pull}125983[#125983]
* Adds top metrics aggregation to AggConfigs, Expressions, and Visualize {kibana-pull}125936[#125936]
* Adds the ability to detach from global time range in *Lens* {kibana-pull}125563[#125563]
* Adds last value, min and max on dates, allow last value on ip_range, number_range, and date_range in *Lens* {kibana-pull}125389[#125389]
* Adds version-aware sorting to data table in *Lens* {kibana-pull}125361[#125361]

> * Adds embeddable enhancement {kibana-pull}125299[#125299]

Reviewer 4: Can be removed

* Cancel discarded searches in *Timelion* {kibana-pull}125255[#125255]
* Cancel discarded searches in *TSVB* {kibana-pull}125197[#125197]
* Adds the ability to allow users to disable auto-apply in *Lens* {kibana-pull}125158[#125158]
* Adds Filter custom label for kibanaAddFilter in *Vega* {kibana-pull}124498[#124498]
* Adds metric Viz config options, title position, and sizing in *Lens* {kibana-pull}124124[#124124]
* Adds the ability to make graph edges easier to click {kibana-pull}124053[#124053]

> * Adds new autoExtendBounds setting to histogram agg config {kibana-pull}118855[#118855]

Reviewer 4: That's rather technical, let's go with `Add "show empty rows" option to intervals function in Lens`

Machine Learning::

> * Combines annotations into one block if multiple of them overlap {kibana-pull}128782[#128782]

Reviewer: ```suggestion
* Combines annotations into one block if multiple annotations overlap {kibana-pull}128782[#128782]
```

> * Adds `throughput_last_minute` to the Deployment stats {kibana-pull}128611[#128611]

Reviewer: ```suggestion
* Adds `throughput_last_minute` to the deployment stats {kibana-pull}128611[#128611]
```

* Support custom icons in maps {kibana-pull}113144[#113144]

Observability::
* Guided setup progress {kibana-pull}128382[#128382]
* Enable check for public beta {kibana-pull}128240[#128240]
* Guided setup button on the overview page {kibana-pull}128172[#128172]
* Show warning when users exceed a Synthetics Node throttling limits {kibana-pull}127961[#127961]
* Adds logging to Inventory Threshold Rule {kibana-pull}127838[#127838]
* O11y rules page {kibana-pull}127406[#127406]
* Enrich documents generated by the synthetics service with `port` information {kibana-pull}127180[#127180]
* Make UI indices space aware (support for spaces) {kibana-pull}126176[#126176]
* Setting for default env for service inventory {kibana-pull}126151[#126151]
* Alerts in overview page {kibana-pull}125337[#125337]
* Adds log rate to Exploratory View {kibana-pull}125109[#125109]
* Support switching between log source modes {kibana-pull}124929[#124929]
* Overview style updates {kibana-pull}124702[#124702]
* Adds full screen/copy button ability in browser inline script editing {kibana-pull}124500[#124500]
* Update position of legend and it's controls {kibana-pull}115854[#115854]

Platform::

> Allow customizing ES client maxSockets {kibana-pull}126937[#126937]

Reviewer 7: ```suggestion
Allow customizing {es} client maxSockets {kibana-pull}126937[#126937]
```

* Remove opacity for fitting line series {kibana-pull}127176[#127176]
* Handle line/area fitting function when the editor has started with bar configuration in *Visualize* {kibana-pull}126891[#126891]
* Preserve custom label when changing with multi-terms settings in *Lens* {kibana-pull}126773[#126773]
* Fixes multi terms fields validation in *Lens* {kibana-pull}126618[#126618]
* Make Embeddable resilient when toggling actions in *Lens* {kibana-pull}126558[#126558]
* Make graph app resilient to no fields or missing data views {kibana-pull}126441[#126441]
* Fixes Formula to Quick functions does not preserve custom formatting in *Lens* {kibana-pull}124840[#124840]
* Inspector displays only visible content {kibana-pull}124677[#124677]
* Coloring tooltips in Heatmap are not properly positioned in *Visualize* {kibana-pull}124507[#124507]
* Adds rison helper and URL encoding for drilldown urls in *TSVB* {kibana-pull}124185[#124185]

Machine Learning::
* Fixes alignment of Anomaly Explorer swim lane annotations label on Firefox {kibana-pull}130274[#130274]
* Fixes Single Metric Viewer chart failing to load if no points during calendar event {kibana-pull}130000[#130000]

> * Fixes Single Metric Viewer for jobs that hasn't been run {kibana-pull}129063[#129063]

Reviewer: ```suggestion
* Fixes Single Metric Viewer for jobs that haven't been run {kibana-pull}129063[#129063]
```

> * Data Frame Analytics: Fix Outlier detection results exploration color legend display {kibana-pull}129058[#129058]

Reviewer: ```suggestion
* Fix outlier detection results exploration color legend display {kibana-pull}129058[#129058]
```

> * Fixes new AD job from saved search with no query filter {kibana-pull}129022[#129022]

Reviewer: ```suggestion
* Fixes new anomaly detection job from saved search with no query filter {kibana-pull}129022[#129022]
```

> * Fixes DFA map saved object sync warning {kibana-pull}128876[#128876]

Reviewer: ```suggestion
* Fixes data frame analytics map saved object sync warning {kibana-pull}128876[#128876]
```

* Fixes lens region map visualization throws a silent error {kibana-pull}129608[#129608]
* Fixes double click issue when deleting a shape {kibana-pull}124661[#124661]

Monitoring::
* Exclude Malwarescore + Ransomware EP alerts from DRule telemetry {kibana-pull}130233[#130233]
* Rename "APM & Fleet Server" to "Integrations Server" {kibana-pull}128574[#128574]
* Fixes sorting by node status on nodes listing page {kibana-pull}128323[#128323]

Observability::
* Service environment should be selected when you edit the agent configuration {kibana-pull}129929[#129929]
* Adds migration to include synthetics and heartbeat indices on 8.2.0 {kibana-pull}129510[#129510]
* Rules summary on the Alerts view is not showing the count of rules {kibana-pull}129052[#129052]
* Fixes shadow for overview panels {kibana-pull}128878[#128878]
* Ensure rum_allow_origins setting only saves valid YAML strings {kibana-pull}128704[#128704]
* Standardize NOW as startedAt from executor options {kibana-pull}128020[#128020]
* Fixes synthetics recorder file upload {kibana-pull}127614[#127614]
* Service maps popover detail metrics are aggregates over all transaction types {kibana-pull}125580[#125580]

Platform::

> * Duplicated datatable available as inspector data for Heatmap chart {kibana-pull}126786[#126786]

Reviewer 7: The title for the original PR here was confusingly worded. This isn't my team (cc @flash1293), but it looks like there are really two things that were fixed here:
- We fixed an issue that would cause duplicate data in the inspector datatable in Lens for Heatmap charts
- We fixed an issue that would cause stale data to persist in the inspector datatable when layers were removed
